src/main/java/csdev/couponstash/model/coupon/PromoCode.java

```diff
@@ -0,0 +1,40 @@
+package csdev.couponstash.model.coupon;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Coupon's promo code in the CouponStash.
+ * Guarantees: immutable
+ */
+public class PromoCode {
+
```

Comment:
Good consideration to not limit the promo code to any validation regex!

Reply:
That's cuz I dunno regex :)